src/api/spec/features/beta/webui/maintenance_workflow_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'browser_helper'
+
+RSpec.describe 'MaintenanceWorkflow', js: true, vcr: true do
+  let(:admin_user) { create(:admin_user) }
+  let(:maintenance_project) do
+    create(:maintenance_project,
+           name: 'MaintenanceProject',
+           title: 'official maintenance space',
+           target_project: [update_project, another_update_project],
+           maintainer: maintenance_coord_user)
+  end
+  let(:user) { create(:confirmed_user, :with_home, login: 'tom') }
+  let(:maintenance_coord_user) { create(:confirmed_user, :with_home, login: 'maintenance_coord') }
+  let(:project) { create(:project_with_repository, name: 'openSUSE:11.4') }
+  let(:another_project) { create(:project_with_repository, name: 'openSUSE:11.5') }
+  let(:branched_project) { create(:project_with_repository, name: 'home:tom:branches:Update') }
+  let(:package) { create(:package_with_file, project: project, name: 'cacti') }
+  let!(:branched_package) { create(:package_with_file, project: branched_project, name: 'cacti.openSUSE_11.4_Update', file_name: 'DUMMY_FILE', file_content: 'boo#12345') }
+  let(:another_package) { create(:package_with_file, project: another_project, name: 'cacti') }
+  let!(:another_branched_package) { create(:package_with_file, project: branched_project, name: 'cacti.openSUSE_11.5_Update', file_name: 'DUMMY_FILE', file_content: 'boo#12345') }
+  let(:update_project) { create(:update_project, maintained_project: project, name: "#{project}:Update") }
+  let(:another_update_project) { create(:update_project, maintained_project: another_project, name: "#{another_project}:Update") }
+  let(:bs_request_without_patchinfo) do
+    create(:bs_request_with_maintenance_incident_actions, source_project_name: branched_project.name, source_package_names: [package.name], target_project_name: maintenance_project.name,
+                                                          target_releaseproject_names: [update_project.name, another_update_project.name])
+  end
+
+  before do
+    User.session = admin_user
+    create(:maintenance_project_attrib, project: maintenance_project)
+  end
+
+  context 'maintenance request without patchinfo' do
+    before do
+      visit request_show_path(bs_request_without_patchinfo)
+    end
+
+    it 'displays information on type of request' do
+      expect(page).to have_text('This is a Maintenance Incident')
+    end
+
+    context 'when accepting request' do
+      before do
+        login(maintenance_coord_user)
+
+        visit request_show_path(bs_request_without_patchinfo)
+        fill_in('reason', with: 'really? ok')
+
+        click_button('Accept request')
+      end
+
+      it 'succeeds' do
+        expect(page).to have_text("Request #{bs_request_without_patchinfo.number} accepted")
+      end
+
+      it 'creates maintenance incident project' do
+        expect(page).to have_text('MaintenanceProject:0')
```

Comment:
You can also expect *real* changes, not only what is displayed. For instance that the request changes the state, that the `:0` project is created etc.